model/src/main/java/org/hibernate/performance/search/model/asset/AutomaticIndexingState.java

```diff
@@ -43,98 +46,95 @@ public AutomaticIndexingState(RelationshipSize relationshipSize, int initialInde
 		}
 	}
 
-	public synchronized void start() {
+	public synchronized void startTrial() {
 		if ( started ) {
 			return;
 		}
 		sessionFactory = HibernateORMHelper.buildSessionFactory( additionalProperties );
-		DomainDataFiller domainDataFiller = new DomainDataFiller( sessionFactory, relationshipSize );
-		for ( int i = 0; i < initialIndexSize; i++ ) {
-			domainDataFiller.fillData( i );
-		}
-		indexInsertPartitions = createInsertPartitions();
-		started = true;
+		start();
 	}
 
-	public synchronized void stop() {
+	public synchronized void stopTrial() {
 		if ( !started ) {
 			return;
 		}
 		if ( sessionFactory != null ) {
 			sessionFactory.close();
 		}
-		indexInsertPartitions = null;
-		indexUpdatePartitions = null;
-		indexDeletePartitions = null;
-		started = false;
+		stop();
 	}
 
-	public AutomaticIndexingInsertPartitionState getInsertPartition(int threadNumber) {
-		checkThreadNumber( threadNumber );
-		return indexInsertPartitions.get( threadNumber );
+	public synchronized void startIteration() {
+		if ( started ) {
+			return;
+		}
+		start();
```

Comment:
I think that it is executed if we start iteration : iteration number > 1. Do you agree?

Reply:
Ah, I see, you call `stop` after each iteration. `start` doesn't start Hibernate Search, it generates partitions and such.

In that case, I'm not entirely sure why you call `start` in `startTrial()` and `stop` in `stopTrial`?

Reply:
You're right we can call it only from `startIteration` and `stopIteration`, since it seems that `startIteration` is always called after `startTrial` and `stopIteration` is always called just before `stopTrial`.